### PR TITLE
#optimization + Renaming + Keyword.all_keys_valid? & co.

### PR DESCRIPTION
**Pull request by CyrusNuevoDia**

### CODE REVIEW

**Comment by CyrusNuevoDia** [comment]
Just added `Keyword.all_keys_valid?` (Issue #19), `Keyword.validate_keys`, and `Keyword.validate_keys!`, and migrated `List.to_sentence` to using the generalized Keyword functions.

File: lib/crutches/keyword.ex [hunk 1]
```diff
@@ -0,0 +1,67 @@
+defmodule Crutches.Keyword do
+  @doc ~S"""
+  Checks a keyword list for unexpected keys, by using a default list of keys. When a bad key is detected it returns false, else it returns true.
```
Comment (mykewould) [inline lib/crutches/keyword.ex]: Can we break up this line so it's under 80 chars (and on line 22)? I know the issue didn't do this, unfortunately Markdown was giving me a hassle. I apologize for that.

**Comment by knrz** [comment]
Done :smiley:

File: lib/crutches/keyword.ex (hunk 1 above)
Comment (CyrusNuevoDia) [inline lib/crutches/keyword.ex]: Done :smiley:

**Comment by mykewould** [comment]
Awesome! Thanks!

**Pull request merged**